File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        check = defaultdict(int)

        for i in range(len(nums)):
            diff = target-nums[i]
            if diff in check:
                return [check[diff],i]
            check[nums[i]] = i

        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const set = new Set()

    for(let i =0; i<nums.length; i++){
        check = target - nums[i]
        if(set.has(check)){
            return [i, nums.indexOf(check)]
        }
        set.add(nums[i])
    }
};